src/runtime/Object.cpp

```diff
@@ -541,7 +541,8 @@ bool Object::defineOwnProperty(ExecutionState& state, const ObjectPropertyName&
                 // Convert the property named P of object O from a data property to an accessor property.
                 // Preserve the existing values of the converted property’s [[Configurable]] and [[Enumerable]] attributes
                 // and set the rest of the property’s attributes to their default values.
-                newDesc = ObjectPropertyDescriptor(desc.value(), (ObjectPropertyDescriptor::PresentAttribute)f);
+                Value val = desc.isValuePresent() ? desc.value() : Value();
+                newDesc = ObjectPropertyDescriptor(val, (ObjectPropertyDescriptor::PresentAttribute)f);
```

Comment:
style: I'd simply inline `val` to avoid one-shot temps, i.e.:
```suggestion
                newDesc = ObjectPropertyDescriptor(desc.isValuePresent() ? desc.value() : Value(), (ObjectPropertyDescriptor::PresentAttribute)f);
```